docs/ACCESSIBILITY.md

```diff
@@ -127,6 +129,61 @@ When the "New messages" separator is activated, it logically moves the separator
 
 Lastly, we style the "New messages" separator like a normal button, styled it to float on top of the transcript, and added `click` and `keypress` event listener for <kbd>ENTER</kbd> and <kbd>SPACEBAR</kbd> key for [activation](https://www.w3.org/TR/wai-aria-practices-1.1/#button).
 
+## UX: Message ordering
+
+### User story
+
+Azure Bot Services is a distributed system and message order is not guaranteed. Web Chat use insertion sort based on the timestamp to order messages.
+
+Messages with a latter timestamp may arrive sooner than messages with a former timestamp. Thus, messages with latter timestamp could appear on the screen first. Then, messages with a former timestamp will get inserted before it.
+
+Because the time between the insertion is usually very short (adjacent packet in a Web Socket connection), users may not see the insertion visually. But screen reader always read messages in the order they appear on the screen, regardless of their positions in the DOM tree. Thus, message order could be confusing to users who relies on screen reader.
+
+![Direct Line sequence diagram](https://raw.githubusercontent.com/microsoft/BotFramework-WebChat/master/docs/media/direct-line-sequence-diagram.png)
+
+### Implementations
+
+We will rectify message order using the `replyToId` property.
+
+`replyToId` is a property set by the Bot Framework SDK and it references the activity the bot is replying to. Web Chat use the `replyToId` property as a hint when rectifying message order.
+
+-  When a message with a `replyToId` property arrive, Web Chat will check if it received the activity with the specified ID:
+   -  If an activity were received with the specified ID, Web Chat will render the activity immediately
+   -  If no activities were received with the specified ID, Web Chat will wait up to 5 seconds for the referencing activity to arrive
+      -  If the activity arrive within 5 seconds, Web Chat will render the activity in the same render loop
+      -  If the activity did not arrive within 5 seconds, Web Chat will render the activity
+-  When a message without a `replyToId` property arrive, Web Chat will render the activity immediately
+
+## UX: Live region transcript
+
+### User story
+
+In a [live region](https://www.w3.org/TR/wai-aria-1.1/#live_region_roles), it is difficult to control which part to be read or excluded from the screen reader.
+
+And sometimes, browser or screen reader could be buggy on how the live region is getting read. For example:
```

Comment:
```suggestion
Also, browsers and screen readers can be inconsistent on reading the live region. For example:
```

Reply:
When I slack for a bit, my English turns really bad. 🤦🏻‍♂️

Reply:
Hahaha it happens